### PR TITLE
Fallback to v1 engine for pagination

### PR DESCRIPTION
**Pull request by MaxKsyunz**
### Description
At the moment, some queries can only be paginated by the legacy engine.
Add tests to confirm fallback happens in those scenarios.
 
Fallback logic already exists in the target branch so this PR adds test coverage.
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: integ-test/src/test/java/org/opensearch/sql/legacy/CursorIT.java [hunk 1]
```diff
@@ -465,6 +483,13 @@ public String executeFetchAsStringQuery(String query, String fetchSize, String r
     return responseString;
   }
 
+  private void verifyIsV1Cursor(String cursor) {
+    if (cursor.isEmpty()) {
+      return;
+    }
+    assertTrue("The cursor '" + cursor + "' is not from v1 engine.", cursor.startsWith("d:"));
```
Comment (Yury-Fridlyand) [inline integ-test/src/test/java/org/opensearch/sql/legacy/CursorIT.java:490]: In legacy cursor can starts with `d:`, `j:` and `a:` https://github.com/Bit-Quill/opensearch-project-sql/blob/114e02a20f66fcab38b5df0c087ba1792da2f2c5/legacy/src/main/java/org/opensearch/sql/legacy/cursor/CursorType.java#L19-L21
```suggestion
    assertTrue("The cursor '" + cursor + "' is not from v1 engine.", !cursor.startsWith("n:"));
```

File: integ-test/src/test/java/org/opensearch/sql/sql/PaginationIT.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  private void verifyIsV1Cursor(JSONObject response) {
+    if (!response.has("cursor")) {
+      return;
+    }
+
+    var cursor = response.getString("cursor");
+    if (cursor.isEmpty()) {
+      return;
+    }
+    assertTrue("The cursor '" + cursor + "' is not from v2 engine.", cursor.startsWith("d:"));
```
Comment (Yury-Fridlyand) [inline integ-test/src/test/java/org/opensearch/sql/sql/PaginationIT.java:69]:
```suggestion
    assertTrue("The cursor '" + cursor + "' is not from v2 engine.", !cursor.startsWith("n:"));
```

File: integ-test/src/test/java/org/opensearch/sql/sql/PaginationIT.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    loadIndex(Index.BEER);
+    loadIndex(Index.ONLINE);
+  }
+
+  @Test
+  public void testSmallDataSet() throws IOException {
+    var query = "SELECT * from " + TEST_INDEX_CALCS;
+    var response = new JSONObject(executeFetchQuery(query, 4, "jdbc"));
+    assertTrue(response.has("cursor"));
+    assertEquals(4, response.getInt("total"));
```
Comment (Yury-Fridlyand) [inline integ-test/src/test/java/org/opensearch/sql/sql/PaginationIT.java:30]: There is a bug in v2 in pagination, that we have to fix. `total` should be 17 (`= count(*)`), but `size` should be 4 (`= fetch_size`)

Comment (Yury-Fridlyand) [reply integ-test/src/test/java/org/opensearch/sql/sql/PaginationIT.java:30]: I think we need a test for these values on all pages

File: opensearch/src/main/java/org/opensearch/sql/opensearch/storage/scan/OpenSearchIndexScanQueryBuilder.java [hunk 4]
```diff
@@ -39,8 +39,9 @@
 @VisibleForTesting
 class OpenSearchIndexScanQueryBuilder extends TableScanBuilder {
   @Override
-  public void pushDownOffset(int i) {
+  public boolean pushDownOffset(int i) {
```
Comment (Yury-Fridlyand) [inline opensearch/src/main/java/org/opensearch/sql/opensearch/storage/scan/OpenSearchIndexScanQueryBuilder.java:42]: `pushDownOffset` was removed in #233
https://github.com/Bit-Quill/opensearch-project-sql/pull/233/files#diff-3334dc7821335d1c35cb35e3da00960106b6ac04c4ca092937c51c076056d272

File: integ-test/src/test/java/org/opensearch/sql/legacy/CursorIT.java (hunk 1 above)
Comment (MaxKsyunz) [reply integ-test/src/test/java/org/opensearch/sql/legacy/CursorIT.java:490]: Legacy only ever produces default cursors. Aggregation and Join enum values are not used.

Comment (MaxKsyunz) [reply integ-test/src/test/java/org/opensearch/sql/legacy/CursorIT.java:490]: Addressed in https://github.com/Bit-Quill/opensearch-project-sql/pull/245/commits/cb97721147571ebad13c0017bb56851ef35b946d

File: integ-test/src/test/java/org/opensearch/sql/sql/PaginationIT.java (hunk 2 above)
Comment (MaxKsyunz) [reply integ-test/src/test/java/org/opensearch/sql/sql/PaginationIT.java]: Addressed in https://github.com/Bit-Quill/opensearch-project-sql/pull/245/commits/cb97721147571ebad13c0017bb56851ef35b946d

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by matthewryanwells — APPROVED** [review]

**Pull request merged**